apps/cowswap-frontend/src/modules/injectedWidget/hooks/useInjectedWidgetPalette.ts

```diff
@@ -7,17 +7,27 @@ import { useLocation } from 'react-router-dom'
 // The theme palette provided by a consumer
 export function useInjectedWidgetPalette(): Partial<CowSwapWidgetPaletteParams> | undefined {
   const { search } = useLocation()
+  const [paletteParams, setPaletteParams] = useState<CowSwapWidgetPaletteParams | undefined>(undefined)
 
-  return useMemo(() => {
+  useEffect(() => {
```

Comment:
Why change to useEffect?

AFAICT, the fix is unrelated, no?

Reply:
As you can see we exit from the effect body when:
```
if (!palette) return undefined
```

Before, when it was `useMemo()` we changed the state to undefined (actually reseted it)

Reply:
So the main change here - we don't set state to undefined when the query parameter `palette` is not set.
We only do that when it has `null` value

Reply:
Got it, thanks for explaining